docs/typescript/reference/dbos.md

```diff
@@ -0,0 +1,953 @@
+  @DBOS.workflow()
+  static async processWorkflow(value: string) {
+    ...
+  }
+
+  // Mark an instance method as a DBOS workflow
+  @DBOS.workflow()
+  processInstanceWorkflow(value: string) {
+    ...
+  }
+
+  // Needed to assign an instance name, see "Instance Methods" below
+  constructor(name: string) {
+    super(name);
+  }
+
+  // Initialize an instance, see "Instance Methods" below
+  async initialize() {
+    return Promise.resolve();
+  }
+}
+
+// Create and register an object instance for executing workflows
+const workflowInstance = DBOS.configureInstance(Workflows, 'InstanceA');
+```
+
+#### Calling Workflow Functions
+Functions that are registered as workflows with `@DBOS.workflow` can be called directly like regular async functions, but will receive DBOS execution guarantees.
+
+```typescript
+// Run functions directly
+const result1 = await Workflows.processWorkflow("value1");
+const result2 = await workflowsInstance.processInstanceWorkflow("value2");
+```
+
+##### Starting Background Workflows
+Workflow functions can also be started asynchronously using `DBOS.startWorkflow`.  When run with `DBOS.startWorkflow`, the function will run in the background, and a `WorkflowHandle` is returned from `DBOS`.  This [handle](../reference/workflow-handles.md) can be used to [interact with the workflow](#interacting-with-workflows) or await its result with `getResult()`.
+
+```typescript
+DBOS.startWorkflow<T extends ConfiguredInstance>(targetInstance: T): InvokeFunctionsAsyncInst<T>;
+DBOS.startWorkflow<T extends object>(targetClass: T): InvokeFunctionsAsync<T>;
+```
+
+Examples:
+```typescript
+// Run static functions asynchronously and get the result later
+const handle3 = await DBOS.startWorkflow(Workflows).processWorkflow("value3");
+// ...
+const result3 = await handle3.getResult();
+
+// Run instance functions asynchronously and get the result later
+const handle4 = await DBOS.startWorkflow(workflowInstance).processInstanceWorkflow("value4");
+// ...
+const result4 = await handle4.getResult();
+
+```
+
+##### Using Workflow Queues
+By default, the `startWorkflow` method [described above](#starting-background-workflows) always starts the target workflow immediately.  If it is desired to control the concurrency or rate of workflow invocations, a [`queue`](../reference/workflow-queues.md) can be specified.
+
+```typescript
+DBOS.withWorkflowQueue<R>(wfq: string, callback: ()=>Promise<R>) : Promise<R>
+```
+
+The `DBOS.withWorkflowQueue` function runs the provided `callback` function and returns its result.  Any workflows started within `callback` will be run on the queue named by `wfq`.
+
+Example:
+```typescript
+// Create a workflow queue for use below
+const wfq = new WorkflowQueue("example_queue", 10, {limitPerPeriod: 50, periodSec: 30});
+
+// Run a workflow on the queue, and await the result
+const qres = await DBOS.withWorkflowQueue(wfq.name, async ()=>{
+  return await Workflows.processWorkflow("value q");
```

Comment:
So does this mean that the callback function can be anything and the return value is not necessarily a handle? Can we make `withWorkflowQueue` the same as `startWorkflow`?

Reply:
In this case, the workflow is not even running on a queue. It's not rate limited or processed asynchronously.

Reply:
I have in my TODO notes that startWorkflow could take more arguments, as it does in Python, including the queue.  But doing it this way, you could run a whole list of things, and any workflows would be run on queues, so I thought it was a nice addition.

Reply:
It should be running on a queue.

https://github.com/dbos-inc/dbos-transact-ts/blob/35ae800e1427051f406999e8818b8c73356ba3cc/tests/contextfreeinst.test.ts#L222

If not, we can fix it.  It's a Promise, so it's processed asynchronously.

Reply:
Parameters added to startWorkflow.